addons/web/static/src/model/relational_model/dynamic_list.js

```diff
@@ -221,6 +221,7 @@ export class DynamicList extends DataPoint {
         } else {
             resIds = await this.getResIds(true);
```

Comment:
I would do it in getResIds directly, so this part would become
```js
        if (records.length) {
            resIds = unique(records.map((r) => r.resId));
        } else {
            resIds = await this.getResIds(true);
        }
```

Reply:
same for _deleteRecords, and nothing needs to be done in _toggleArchive then

Reply:
unique comes form utils/arrays.js, but feel free to keep doing it manually, I don't really mind